src/concurrency/sync.rs

```diff
@@ -44,7 +47,20 @@ macro_rules! declare_id {
 }
 pub(super) use declare_id;
 
-declare_id!(MutexId);
+#[derive(Default, Clone, Debug)]
+pub struct MutexRef(Rc<RefCell<Mutex>>);
```

Comment:
Please define this right after `struct Mutex` so that related things are close together.